### PR TITLE
ceph: add service account for mon pods

### PR DESCRIPTION
**Pull request by tareqhs**
Signed-off-by: Tareq Sharafy <tareq.sha@gmail.com>

Solves https://github.com/rook/rook/issues/6673

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [X] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [X] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Review by travisn — CHANGES_REQUESTED** [review]
If we're going to create this for mons, we should be consistent across all the daemons that are not consistently using a service account. For example, mon, mgr, rgw, rbdmirror, and nfs daemons all use the default SA. Therefore, I suggest the following:
- Name the new SA `rook-ceph-default`
- Add the SA to all the daemons that currently don't specify an SA
- The SA needs to be defined in three places: common.yaml, helm chart (you already have this location), and the integration tests (ceph_manifests.go).

File: cluster/examples/kubernetes/ceph/common.yaml [hunk 1]
```diff
@@ -1125,3 +1125,10 @@ roleRef:
   name: rbd-external-provisioner-runner
   apiGroup: rbac.authorization.k8s.io
 # OLM: END CSI RBD CLUSTER ROLEBINDING
+---
+# Service account for other components
```
Comment (travisn) [inline cluster/examples/kubernetes/ceph/common.yaml:1129]: We will need some OLM tags around the SA, similar to other SAs in common.yaml. 
```suggestion
# OLM: BEGIN DEFAULT SERVICEACCOUNT
# Service account for other components
...
# OLM: END DEFAULT SERVICEACCOUNT
```

Then we will need to add the new tag to the generation script [here](https://github.com/rook/rook/blob/master/cluster/olm/ceph/generate-rook-csv.sh).

**Review by travisn — CHANGES_REQUESTED** [review]
The integration tests are failing. The logs show that the mon canary pod isn't starting. Were you able to test the changes locally and see the mons start?
```
op-mon: sched-mon: waiting for canary pod creation rook-ceph-mon-a-canary
```

If you could also update the [ceph-prerequisites.md](https://github.com/rook/rook/blob/master/Documentation/k8s-pre-reqs.md#example-setup-for-a-ceph-cluster) with the name of the new SA instead of the default SA. While you're at it, might be nice to mention the helm chart has a setting for it.

**Comment by leseb** [comment]
@tareqhs please address the conflicts, any plan to pursue this? Thanks

**Comment by travisn** [comment]
@tareqhs Just to confirm, are you still planning on it? If not, we'll find a new owner. thanks

**Comment by parth-gr** [comment]
Hy @travisn, I would like to work upon this [issue](https://github.com/rook/rook/issues/6673) and complete this **[PR](https://github.com/rook/rook/pull/6674)** :)

**Comment by travisn** [comment]
This work has been moved to #7393.

**Comment by parth-gr** [comment]
@edit: moved to [#7406](https://github.com/rook/rook/pull/7406) from [#7393](https://github.com/rook/rook/pull/7393).

**Pull request closed without merging**